### PR TITLE
tests: replace assert.Equal on proto with protoassert 

### PR DESCRIPTION
**Pull request by janisz**
## Description

This PR uses `eg` to automatically replace equality assertions in tests with proto aware replacements. 

```go
package template

import (
	"github.com/golang/protobuf/proto"
	"github.com/stackrox/rox/pkg/protoassert"
	"github.com/stretchr/testify/assert"
	"testing"
)

func before(t *testing.T, a, b proto.Message) bool {
	return assert.Equal(t, a, b)
}

func after(t *testing.T, a, b proto.Message) bool {
	return protoassert.Equal(t, a, b)
}
```

```go
package template

import (
	"github.com/golang/protobuf/proto"
	"github.com/stackrox/rox/pkg/protoassert"
	"github.com/stretchr/testify/suite"
)

func before(s *suite.Suite, a, b proto.Message) bool {
	return s.Equal(a, b)
}

func after(s *suite.Suite, a, b proto.Message) bool {
	return protoassert.Equal(s.T(), a, b)
}
```

```go
package template

import (
	"github.com/golang/protobuf/proto"
	"github.com/stackrox/rox/pkg/protoassert"
	"github.com/stretchr/testify/require"
	"testing"
)

func before(t *testing.T, a, b proto.Message) {
	require.Equal(t, a, b)
}

func after(t *testing.T, a, b proto.Message) {
	protoassert.Equal(t, a, b)
}
```

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Comment by janisz** [comment]
- #11531
- #11533
- #11503
- #11507

File: sensor/upgrader/plan/preserve_test.go [hunk 1]
```diff
@@ -152,7 +153,7 @@ func TestPreserveResources(t *testing.T) {
 	var mergedDS v1.DaemonSet
 	require.NoError(t, convert(scheme.Scheme, newDSUnstructured, &mergedDS))
 
-	assert.Equal(t, expectedMergedDS, &mergedDS)
+	protoassert.Equal(t, expectedMergedDS, &mergedDS)
```
Comment (rhybrillou) [inline sensor/upgrader/plan/preserve_test.go:156]: This type is a k8s.io type, none of our generated types (as far as I can tell). It matches the `proto.Message` interface, though.

File: sensor/upgrader/plan/preserve_test.go [hunk 2]
```diff
@@ -249,7 +250,7 @@ func TestPreserveTolerations(t *testing.T) {
 	var mergedDeploy v1.Deployment
 	require.NoError(t, convert(scheme.Scheme, newDeployUnstructured, &mergedDeploy))
 
-	assert.Equal(t, expectedMergedDeploy, &mergedDeploy)
+	protoassert.Equal(t, expectedMergedDeploy, &mergedDeploy)
```
Comment (rhybrillou) [inline sensor/upgrader/plan/preserve_test.go:253]: same as the previous v1 type in this file.

File: central/vulnmgmt/vulnerabilityrequest/datastore/datastore_impl_test.go [hunk 3]
```diff
@@ -481,7 +482,7 @@ func (s *VulnRequestDataStoreTestSuite) TestUpdateRequestExpiryUpdatesExpiryIfRe
 	resp, err := s.datastore.UpdateRequestExpiry(authn.ContextWithIdentity(requesterWriteSac, s.mockIdentity, s.T()), "id", "update comment", expiry)
 	s.NoError(err)
 
-	s.Equal(expiry, resp.GetUpdatedDeferralReq().GetExpiry())
+	protoassert.Equal(s.T(), expiry, resp.GetUpdatedDeferralReq().GetExpiry())
 	s.Equalf(req.GetReq(), resp.GetReq(), "original request should not have been updated")
```
[the remark this answers is not in the document]
Comment (janisz) [reply central/vulnmgmt/vulnerabilityrequest/datastore/datastore_impl_test.go:486]: Right, I'd create a separated PR for `Equalf` family

[the remark this answers is not in the document]
Comment (janisz) [reply central/vulnmgmt/vulnerabilityrequest/datastore/datastore_impl_test.go:486]: - https://github.com/stackrox/stackrox/pull/11618

File: pkg/search/parser/url_parser_test.go
[the remark this answers is not in the document]
Comment (janisz) [reply pkg/search/parser/url_parser_test.go:1]: - https://github.com/stackrox/stackrox/pull/11619

**Pull request merged**